logic/src/commonMain/kotlin/com/wire/kalium/logic/data/auth/login/LoginRepository.kt

```diff
@@ -13,23 +14,23 @@ import com.wire.kalium.network.utils.httpResponseCookies
 import com.wire.kalium.network.utils.isSuccessful
 
 interface LoginRepository {
-    suspend fun loginWithEmail(email: String, password: String, shouldPersistClient: Boolean): Either<CoreFailure, AuthSession>
-    suspend fun loginWithHandle(handle: String, password: String, shouldPersistClient: Boolean): Either<CoreFailure, AuthSession>
+    suspend fun loginWithEmail(email: String, password: String, shouldPersistClient: Boolean, serverConfig: ServerConfig): Either<CoreFailure, AuthSession>
```

Comment:
Why are we passing the `serverConfig` for every method which makes a request now? Wasn't the idea to pass the network config once when creating a `AuthenticationScope` and then forward it to `LoginNetworkContainer`?

Reply:
this is **only** for login since kalium hold no info about the BE before calling `LoginUseCase`

Reply:
But why not hold the information about the BE? I'm not saying we should persist it, just hold it in memory for the duration off the `AuthenticationScope`.

Reply:
@typfel IIRC, I discussed this with @MohamadJaara. One problem with this would be dependency injection in the apps. The ViewModels would not be able to just get a `LoginUseCase`, they would actually need to get `CoreLogic` and use it directly to get a `AuthenticationScope`.


One way to fix this would be by wrapping the `UseCase` in the app.

```kotlin
class MyAppLoginUseCase(private val coreLogic: CoreLogic) {

    suspend fun invoke(backendStuff: BackendSettings, email: String, password: String) : Result {
        coreLogic.authenticationScope(backendStuff) {
            loginUsingEmailAndPassword(email, password)
        }
    }
}
```

A bit more annoying for all the use cases inside `AuthenticationScope`.


I didn't find have any favourites in this.

There might be some UseCases inside `AuthenticationScope`  that don't require backend at all. Like `getListOfCurrentSessions` or `getCurrentActiveSession` and `setSessionAsActive`.

Maybe we can find another way to unify and reduce the duplication for the ones where backend settings are required. Like making them extend the same `BaseLoginUseCase` that takes backend config, etc.